Launcher/Classes/BashOutput.swift

```diff
@@ -0,0 +1,115 @@
+import AppKit
+
+enum BashOutput {
+
+    private static let colorCodes: [Int: NSColor] = [
+        30: .black,
+        31: .red,
+        32: .green,
+        33: .yellow,
+        34: .blue,
+        35: .magenta,
+        36: .cyan,
+        37: .white,
+        90: .gray,
+        91: .red,
+        92: .green,
+        93: .yellow,
+        94: .blue,
+        95: .magenta,
+        96: .cyan,
+        97: .white
+    ]
+
+    private static let styleCodes: [Int: NSFont] = [
+        0: .systemFont(ofSize: 15),
+        1: .boldSystemFont(ofSize: 15),
+    ]
+
+    private static func code(forKey key: Int) -> Any? {
+        if let colorCode = colorCodes[key] {
+            return colorCode
+        } else if let styleCode = styleCodes[key] {
+            return styleCode
+        }
+        return nil
+    }
+
+    struct ANSIGroup {
+        let codes: [Int]
+        let string: String
+    }
+
+    private static func parseExistingANSI(string: String) -> [ANSIGroup] {
+        var results: [ANSIGroup] = []
+        let regexHandler = RegexHandler()
+        let matches = regexHandler.matches(for: "\\u001B\\[([^m]*)m(.+?)\\u001B\\[0m", in: string, global: true)
+
+        for match in matches {
+            let mutlipleMatches = regexHandler.matches(for: "\\u001B\\[([^m]*)m", in: match, global: true)
+            if mutlipleMatches.count > 2 {
+                for i in 0...mutlipleMatches.count - 2 {
+                    let strippedMatches = mutlipleMatches[i]
+                        .replacingOccurrences(of: "\u{1B}[", with: "")
+                        .replacingOccurrences(of: "m", with: "")
+                    let regex = "\\u001B\\[(\(strippedMatches))m(.+?)\\u001B\\[0m"
+                    let multipleParts = regexHandler.matches(for: regex, in: match)
+
+                    if multipleParts.count < 2 {
+                        continue
+                    }
+
+                    let codes = multipleParts[1]
+                        .split { $0 == ";" }
+                        .map(String.init)
```

Comment:
I improved these. Instead of `map { String($0) }` we're passing `String.init` in the `map` now.